src/Domain/Enumeration/ContactMediaEnum.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Domain\Enumeration;
+
+class ContactMediaEnum
+{
+    public const EMAIL = 'email';
+    public const SMS = 'sms';
+    public const PHONE = 'phone';
+    public const FAX = 'fax';
+    public const ADDRESS = 'address';
+    public const FACEBOOK = 'facebook';
+    public const TWITTER = 'twitter';
+    public const GOOGLE = 'google';
+    public const SPOTIFY = 'spotify';
+    public const PINTEREST = 'pinterest';
+    public const TUMBLR = 'tumblr';
+    public const STUMBLE_UPON = 'stumble_upon';
+    public const LINKEDIN = 'linkedin';
+    public const REDDIT = 'reddit';
+    public const DIGG = 'digg';
+    public const INSTAGRAM = 'instagram';
+    public const OTHER = 'other';
+
+    public const VALID_MEDIA = [
```

Comment:
This way of checking valid values differs from how we do it our other enums. Checking a key is faster than checking through an array (O(1) vs O(n)) but we shouldn't try and avoid having multiple ways of doing a single thing.

Having things that consume enums check if a value is valid in differing ways differ based on the internal data structure (`isset(Enum::VALID_MEDIA[$value])` here vs `in_array($value, Enum::validValues())` in other enums) feels a bit janky.

Perhaps the long term solution is to have Enums expose a static isValid method:

```php

private const VALID_VALUES = [/*...*/];

public static isValid($value): bool {
  return isset(self::VALID_VALUES);
}
```


We can then update the other classes to have this isValid convention in a separate PR. What do you think?

Reply:
Even more thoughts: Why do we even need to validate at this phase anyway? Surely any validation would have occurred on ingest in Faucet and we know we've got good information within the database.